packages/core/src/routes/interaction/verifications/profile-verification.ts

```diff
@@ -0,0 +1,219 @@
+import type { Profile, User } from '@logto/schemas';
+import { Event } from '@logto/schemas';
+import { argon2Verify } from 'hash-wasm';
+
+import { getLogtoConnectorById } from '#src/connectors/index.js';
+import RequestError from '#src/errors/RequestError/index.js';
+import {
+  findUserById,
+  hasUser,
+  hasUserWithEmail,
+  hasUserWithPhone,
+  hasUserWithIdentity,
+} from '#src/queries/user.js';
+import assertThat from '#src/utils/assert-that.js';
+
+import { registerProfileSafeGuard } from '../types/guard.js';
+import type {
+  InteractionContext,
+  Identifier,
+  AccountIdIdentifier,
+  SocialIdentifier,
+} from '../types/index.js';
+import { isUserPasswordSet } from '../utils/index.js';
+
+const findUserByIdentifier = async (identifiers: Identifier[]) => {
+  const accountIdentifier = identifiers.find(
+    (identifier): identifier is AccountIdIdentifier => identifier.key === 'accountId'
+  );
+
+  assertThat(
+    accountIdentifier,
+    new RequestError({
+      code: 'session.unauthorized',
+      status: 401,
+    })
+  );
+
+  return findUserById(accountIdentifier.value);
+};
+
+const protectedIdentifierVerification = (
```

Comment:
The function name should be in a verb-object format. For example, `verifyProtectedIdentifier()`

Reply:
udpated